webpack.config.js

```diff
@@ -0,0 +1,109 @@
+var webpack = require("webpack");
+
+// Pass in the build configuration as environment variables
+const CONFIG   = process.env.BUILD_CONFIG
+const PLATFORM = process.env.BUILD_PLATFORM;
+
+// Modify the webpack settings based on the configuration
+var plugins = [];
+var defines = {
+    DEBUG            : false,
+    PLATFORM_NODE    : false,
+    PLATFORM_BROWSER : false,
+};
+
+var bundlePlatform = "";
+var bundleSuffix = "";
+var libraryTarget = "";
+var target = "";
+var devtool = undefined;
+
+switch (CONFIG) {
+    case "debug":
+        defines.DEBUG = true;
+        bundleSuffix = "-debug";
+        devtool = "source-map";
+        break;
+
+    case "prod":
+        plugins.push(new webpack.optimize.UglifyJsPlugin({
```

Comment:
Yes, I believe we do. The short answer in the general case is that minified code (still) tends to run faster on Node than unminified code (an example is React.js), to the best of my understanding at least. In the specific case of this code, It's relying on the dead code elimination of this compilation step to ensure those argument checks truly are zero ovehead (i.e. non-existent!). The newest versions of Node/V8 may be smart enough to do that natively, but `const` actually hasn't been around that long in JS so I would be hesitant to rely on true constant-propogation and code path elimination in older versions of V8.